src/hexer_package.erl

```diff
@@ -23,20 +23,29 @@ publish() ->
    , details := Details
    } = hexer_utils:load_app_info(),
   Deps = hexer_deps:resolve(AppDir),
-  case validate_app_details(Details) of
-    ok -> publish(AppDir, Name, Version, Deps, Details);
-    {error, Error} -> throw(Error)
-  end.
+  ok = validate_app_details(Details),
+  ok = validate_hex_dependencies(Deps),
+  publish(AppDir, Name, Version, Deps, Details).
 
 %%------------------------------------------------------------------------------
 %% Internal Functions
 %%------------------------------------------------------------------------------
+-spec validate_hex_dependencies([tuple()]) -> ok | {error, any()}.
+validate_hex_dependencies(Deps) ->
+  case [{no_hex_dependency, Dep} || {no_hex_dependency, Dep} <- Deps] of
```

Comment:
You can use this chance to get rid of the unneeded atom:
```erlang
case [Dep || {no_hex_dependency, Dep} <- Deps] of
```
So that you don't need to do it below.